scripts/cleanup_edomain_templates.py

```diff
@@ -35,77 +34,35 @@ def make_ssh_client(provider_mgmt):
+    parser.add_argument("--max-templates",
+                        dest="max_templates",
+                        default=5,
                         help="max number of templates to be deleted at a time"
-                             "e.g. --max-templates 6 deletes 6 templates at a time",
-                        default=5)
+                             "e.g. --max-templates 6 deletes 6 templates at a time")
     args = parser.parse_args()
     return args
 
 
-def is_ovirt_engine_running(provider_mgmt):
-    try:
-        with make_ssh_client(provider_mgmt) as ssh_client:
-            stdout = ssh_client.run_command('systemctl status ovirt-engine').output
-            # fallback to sysV commands if necessary
-            if 'command not found' in stdout:
-                stdout = ssh_client.run_command('service ovirt-engine status').output
-        return 'running' in stdout
-    except Exception as e:
-        logger.exception(e)
-        return False
-
-
-def change_edomain_state(provider_mgmt, state, edomain):
-    try:
-        # fetch name for logging
-        provider_name = provider_mgmt.kwargs.get('name')
-        log_args = (provider_name, edomain, state)
-
-        api = provider_mgmt.api
-        dcs = api.datacenters.list()
-        for dc in dcs:
-            export_domain = dc.storagedomains.get(edomain)
-            if export_domain:
-                if state == 'maintenance' and export_domain.get_status().state == 'active':
-                    dc.storagedomains.get(edomain).deactivate()
-                elif state == 'active' and export_domain.get_status().state != 'active':
-                    dc.storagedomains.get(edomain).activate()
-
-                wait_for(is_edomain_in_state, [api, state, edomain], fail_condition=False, delay=5)
-                print('RHEVM:{}, domain {} set to "{}" state'.format(*log_args))
-                return True
-        return False
-    except Exception as e:
-        print(e)
-        print('RHEVM:{} Exception setting domain {} to "{}" state'.format(*log_args))
-        return False
-
-
-def is_edomain_in_state(api, state, edomain):
-    dcs = api.datacenters.list()
-    for dc in dcs:
-        export_domain = dc.storagedomains.get(edomain)
-        if export_domain:
-            return export_domain.get_status().state == state
-    return False
-
-
 # get the domain edomain path on the rhevm
-def get_edomain_path(api, edomain):
-    edomain_id = api.storagedomains.get(edomain).get_id()
-    edomain_conn = api.storagedomains.get(edomain).storageconnections.list()[0]
-    return ('{}/{}'.format(edomain_conn.get_path(), edomain_id),
-            edomain_conn.get_address())
+def get_edomain_path(provider_mgmt, edomain):
+    edomain = provider_mgmt.api.system_service().storage_domains_service().list(
+        search='name={}'.format(edomain),
+        follow='storage_connections')[0]
+    edomain_conn = edomain.storage_connections[0]
```

Comment:
Maybe wrap this in a `Try/Except`?

Reply:
I can add a re-try maybe? The calls to this method are themselves in try/except blocks that keeps the script running on other providers if one fails. Did your local run that hit a ConnectionError die completely, or just log the exception against that one provider?